core/kernel/dt.c

```diff
@@ -925,6 +925,7 @@ int add_res_mem_dt_node(struct dt_descriptor *dt, const char *name,
 	if (IS_ENABLED2(_CFG_USE_DTB_OVERLAY)) {
 		len_size = sizeof(paddr_t) / sizeof(uint32_t);
 		addr_size = sizeof(paddr_t) / sizeof(uint32_t);
+		found = false;
```

Comment:
I would suggest to add an inline comment here, e.g.:
`/* Enforce a reserved-memory node is added in the overlay fragment */`